File: backend/internal/model/node.go
package model

import (
	"time"
)

type Node struct {
	ID             uint           `json:"id" gorm:"primaryKey"`
	SubscriptionID uint           `json:"subscription_id" gorm:"index"`
	Name           string         `json:"name" gorm:"size:256"`
	DisplayName    string         `json:"display_name" gorm:"size:256"`
	NodeType       string         `json:"node_type" gorm:"size:32"`
	Server         string         `json:"server" gorm:"size:256"`
	Port           int            `json:"port"`
	Region         string         `json:"region" gorm:"size:64;index"`
	RawURI         string         `json:"raw_uri" gorm:"type:text"`
	ConfigJSON     JSON           `json:"config_json" gorm:"type:jsonb"`
	Latency        int            `json:"latency"`
	LastCheck      *time.Time     `json:"last_check,omitempty"`
	Status         int8           `json:"status" gorm:"default:1"`
	CreatedAt      time.Time      `json:"created_at"`
}

File: backend-python/models/node.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, JSON
from sqlalchemy.orm import relationship
from .database import Base
from ..utils.time import get_current_time


class Node(Base):
    __tablename__ = "nodes"

    id = Column(Integer, primary_key=True, index=True)
    subscription_id = Column(Integer, ForeignKey("subscriptions.id"), index=True)
    name = Column(String(256))
    display_name = Column(String(256))
    node_type = Column(String(32))
    server = Column(String(256))
    port = Column(Integer)
    region = Column(String(64))
    raw_uri = Column(Text)
    config_json = Column(JSON)
    latency = Column(Integer, default=0)
    status = Column(Integer, default=1)
    created_at = Column(DateTime, default=get_current_time)

    subscription = relationship("Subscription", back_populates="nodes")

    def __repr__(self):
        return f"<Node(id={self.id}, name={self.name}, type={self.node_type})>"
